### PR TITLE
Parallelize processing of Kafka record on topic level

### PR DESCRIPTION
**Pull request by vilunov**

### CODE REVIEW

File: core/src/main/scala/com/evolutiongaming/kafka/flow/ConsumerFlow.scala [hunk 1]
```diff
@@ -84,11 +84,10 @@ object ConsumerFlow {
     def poll = {
       val flowList = flows.toList // optimization, execute toList once instead of on each `consumer.poll`
       consumer.poll(config.pollTimeout) flatTap { consumerRecords =>
-        flowList traverse { case (topic, flow) =>
-          val topicRecords = consumerRecords.values filter { case (partition, _) =>
-            partition.topic == topic
-          }
-          flow(ConsumerRecords(topicRecords))
+        val recordsByTopic = consumerRecords.values.groupBy(_._1.topic)
```
Comment (t3hnar) [inline core/src/main/scala/com/evolutiongaming/kafka/flow/ConsumerFlow.scala:87]:
```
consumerRecords
.values
.groupBy(_._1.topic)
.parFoldMapA { (topic, records) +> } 
```

please, we do not need use `flowList` here, right?

File: core/src/test/scala/com/evolutiongaming/kafka/flow/KafkaFlowSpec.scala [hunk 2]
```diff
@@ -143,6 +145,15 @@ object KafkaFlowSpec {
 
   type F[A] = SyncIO[A]
 
+  // TODO rewrite in IO
+  implicit val parallelForSyncIO = new Parallel[F] {
```
Comment (t3hnar) [inline core/src/test/scala/com/evolutiongaming/kafka/flow/KafkaFlowSpec.scala:149]: there is `Parallel.identity` instead

**Review by t3hnar — CHANGES_REQUESTED** [review]

File: core/src/main/scala/com/evolutiongaming/kafka/flow/ConsumerFlow.scala (hunk 1 above)
Comment (vilunov) [reply core/src/main/scala/com/evolutiongaming/kafka/flow/ConsumerFlow.scala:87]: we still need to call `apply` on flows even when we haven't received record for the topic, as we need to trigger timers for it

Comment (t3hnar) [reply core/src/main/scala/com/evolutiongaming/kafka/flow/ConsumerFlow.scala:87]: than I propose the only change to be done: replace `traverse` with `parFoldMapA`, you'd better leave `filter` inside of parallelised snippet

Comment (vilunov) [reply core/src/main/scala/com/evolutiongaming/kafka/flow/ConsumerFlow.scala:87]: the filter inside is not going to be parallelised, only the IO will be. the function itself is called sequentially

Comment (t3hnar) [reply core/src/main/scala/com/evolutiongaming/kafka/flow/ConsumerFlow.scala:87]: it's more complex than that, you can make a test to check it out.

**Review by t3hnar — APPROVED** [review]

**Pull request merged**